cns/restserver/ipam.go

```diff
@@ -434,26 +563,46 @@ func (service *HTTPRestService) unassignIPConfig(ipconfig cns.IPConfigurationSta
 func (service *HTTPRestService) releaseIPConfig(podInfo cns.PodInfo) error {
 	service.Lock()
 	defer service.Unlock()
-
-	ipID := service.PodIPIDByPodInterfaceKey[podInfo.Key()]
-	if ipID != "" {
-		if ipconfig, isExist := service.PodIPConfigState[ipID]; isExist {
-			logger.Printf("[releaseIPConfig] Releasing IP %+v for pod %+v", ipconfig.IPAddress, podInfo)
-			_, err := service.unassignIPConfig(ipconfig, podInfo)
-			if err != nil {
-				return fmt.Errorf("[releaseIPConfig] failed to mark IPConfig [%+v] as Available. err: %v", ipconfig, err)
+	numIPsToBeReleased := len(service.PodIPIDByPodInterfaceKey[podInfo.Key()])
+	ipsToBeReleased := make([]cns.IPConfigurationStatus, numIPsToBeReleased)
+
+	for i, ipID := range service.PodIPIDByPodInterfaceKey[podInfo.Key()] {
+		if ipID != "" {
+			if ipconfig, isExist := service.PodIPConfigState[ipID]; isExist {
+				ipsToBeReleased[i] = ipconfig
+			} else {
+				logger.Errorf("[releaseIPConfig] Failed to get ipconfig %+v and pod info is %+v. Pod to IPID exists, but IPID to IPConfig doesn't exist, CNS State potentially corrupt",
```

Comment:
shouldn't we return from here? There is no point in moving forward in this func if we cannot mark this assigned ID for releasing.

Reply:
yeah it would make more sense to return from here otherwise it will say that it is successful at the end